Week 4/Enigma-java/src/main/java/myDecryption.java

```diff
@@ -0,0 +1,79 @@
+public class myDecryption {
+
+   // main method
+   public static void main(String[] args) 
+   {
+   
+      // create an instance of the myDecryption class
+      myDecryption decryptCaesar = new myDecryption();
+   
+      // Call the decryptString
+      System.out.println(decryptCaesar.decryptString("ABC", 3));
+   }
+
+   // method to decrypt string using caesar cipher
+   public String decryptString(String msg, int shift) 
+   {
+      // Create an array
+      // int myArray = ["EKMFLGDQVZNTOWYHXUSPAIBRCJ"];
+      // Realised working with letters in a normal String format is more
+      // straightforward than using an array
+   
+      // find the index of each letter
+      // check if character is indeed a letter and if not print an error message
+      // take note of count?
+      // reverse the caesar shift
+      // decrement
+      // shift--;
+   
+      // NB!! All string characters must be in uppercase
+      // -- minus
+      
+      // Define the alphabet
+      String alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+      // String.charAt(0);
+   
+      // Build a stringbuilder to store the decrypted message
+      StringBuilder decryptedMsg = new StringBuilder();
+   
+      // for loop to iterate through the characters in the message
+      for (int i = 0; i < msg.length(); i++) 
```

Comment:
##this will account for non-characters in the string. They will be appended to the final string without being shifted

for (int i = 0; i < msg.length(); i++) 
      {
         char currentChar = Character.toUpperCase(msg.charAt(i));
         
   
         if (Character.isLetter(currentChar))  //if it is a char, shift and append
         {  
            int alphaIndex = alphabet.indexOf(currentChar);
            alphaIndex = (alphaIndex - shift + 26) % 26;
            char decryptedChar = alphabet.charAt(alphaIndex);
             decryptedMsg.append(decryptedChar);
         }
         else 
         {
            decryptedMsg.append(currentChar);         //if it's not a char, just append without shifting
         }
         
         // Decrement the shift for the next character
         shift++;
      }
```suggestion
      for (int i = 0; i < msg.length(); i++) 
```